### DISCUSSION TITLE
Variables cocatenation doesn't work in Github Action Job

### DISCUSSION THREAD

**Discussion by rshellnet** [community/community#54567] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi All,

I'm trying to concatenate two variables and this is not working at all.

This is my job:

name: My Workflow

on: push

jobs:
  my-job:
    runs-on: ubuntu-latest

    steps:
      - name: Another test
        run: echo "aaa=bbb" >> "$GITHUB_ENV"
      - name: Output  
        run: |
          echo $aaa
      - name: Adding second
        run: echo "ccc=ddd" >> "$GITHUB_ENV"
      - name: Concat
        run: |
          echo "eee=${{ env.aaa }}-${{ env.ccc }}" >> $GITHUB_ENV
          echo "${{ env.eee }}"
      - name: Phind example
        env:
          EXISTING_VAR: value1
          OTHER_EXISTING_VAR: value2
        run: |
          echo "NEW_VAR=${{ env.EXISTING_VAR }}/subdir/${{ env.OTHER_EXISTING_VAR }}" >> $GITHUB_ENV
          echo ${{ env.NEW_VAR }}
I got both evn.eee and env.NEW_RAW empty.
Thanks in the advance.

**Comment by airtower-luna** [comment, marked as answer]
There are two reasons here why this might break. The first one *definitely* applies, the second one *might* (and you should fix it either way).

1. When you write a variable to `$GITHUB_ENV` it'll be available in *following* steps. The shell of the *current* step doesn't know anything about it. If you need the new variable only in that step, do a normal shell variable assignment instead. If you need it later, move the commands that use the variable into a separate step.
2. `${{ env.aaa }}` is Actions syntax, which is replaced *verbatim* with the value before the shell sees your commands (that's also the reason for point 1). That means shell escaping won't work right if your variable contains any shell control characters (quotes, semicolons, newlines, …) and might break in funny ways depending on the values in your variables. Withing `run` blocks, you should use shell syntax to access environment variables instead, e.g. `${aaa}`.

**Reply by rshellnet** [reply]
Great, thank you a lot! First option worked as you told.
